packages/babel-parser/src/tokenizer/types.js

```diff
@@ -155,7 +155,7 @@ export const types: { [name: string]: TokenType } = {
   bitShift: createBinop("<</>>/>>>", 8),
   plusMin: new TokenType("+/-", { beforeExpr, binop: 9, prefix, startsExpr }),
   // startsExpr: required by v8intrinsic plugin
-  modulo: new TokenType("%", { beforeExpr, binop: 10, startsExpr }),
```

Comment:
The `beforeExpr` is required by `v8Intrinsic` plugin.

Reply:
Is there a v8intrinsic test case that is broken by this change? The `babel-parser/test/fixtures/v8intrinsic` suite is still succeeding.

It does seem that currently, when babel-parser is configured with both `pipelineOperator: { proposal: "hack", topicToken: "%" }` and `v8intrinsic` on, `x |> %` breaks due to`parseExprAtom` always calling `parseV8Intrinsic` first, which then calls `unexpected`. This seems to be a separate problem, though.

Reply:
I think `beforeExpr` is not needed for the `v8intrinsic` plugin, but for the normal `%` operator.

Could you check if this code is still parsed correctly?
```js
5 % /3/g
```

We might need to manually manage `exprAllowed` for `%` instead of relying on `beforeExpr`.


Also, we should check if these two things are parsed correctly:
```js
a |> %
function f() {}
```

```js
a %
function f() {}
```

Reply:
@nicolo-ribaudo: I added tests for those cases in 5d24958649f0b130cfe31927be00bf3f6254ae44. They seem to be still parsing correctly.

Reply:
Oh awesome. I'll still try to find an example that breaks it because I'm not 100% convinced yet that `beforeExpr` is completely unnecessary, but if I cannot find it then it's good 😅